lib/src/scanner.dart

```diff
@@ -1587,10 +1587,24 @@ class Scanner {
         _inBlockContext,
       SP || TAB || LF || CR || BOM => false,
       NEL => true,
-      _ => _isStandardCharacter(char)
+      _ => _isStandardCharacterAt(offset)
     };
   }
 
+  bool _isStandardCharacterAt(int offset) {
+    var first = _scanner.peekChar(offset);
+    if (first == null) return false;
+
+    if (isHighSurrogate(first)) {
+      var next = _scanner.peekChar(offset + 1);
+      if (next != null && isLowSurrogate(next)) {
+        return _isStandardCharacter(decodeSurrogatePair(first, next));
+      }
+    }
+
+    return _isStandardCharacter(first);
+  }
+
```

Comment:
(Cannot unsee: The comment on `_asHex` below should say "hexadecimal", not "hexidecimal". Now I've said it. Feel free to ignore.)

Reply:
Found 3 occurrences, I'll ignore them in this PR.